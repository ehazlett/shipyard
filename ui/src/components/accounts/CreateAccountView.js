import React from 'react';

import { Container, Grid, Segment } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import CreateAccountForm from './CreateAccountForm';

const CreateAccountView = (props) => (
  <Container>
    <Grid>
      <Grid.Row>
        <Grid.Column width={16}>
          <Segment basic>
            <div className="ui breadcrumb">
              <Link to="/accounts" className="section">Accounts</Link>
              <div className="divider"> / </div>
              <div className="active section">Create a new account</div>
            </div>
          </Segment>
          <Segment basic>
            <CreateAccountForm {...props} />
          </Segment>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

export default CreateAccountView;
