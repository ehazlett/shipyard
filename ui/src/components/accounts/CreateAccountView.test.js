import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router";
import CreateAccountView from "./CreateAccountView";
import renderer from "react-test-renderer";

it("CreateAccountView renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <CreateAccountView />
    </MemoryRouter>,
    div
  );
});

it("CreateAccountView snapshot matches", () => {
  const tree = renderer
    .create(<MemoryRouter><CreateAccountView /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
