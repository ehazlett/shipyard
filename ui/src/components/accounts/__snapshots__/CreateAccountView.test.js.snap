exports[`test CreateAccountView snapshot matches 1`] = `
<div
  className="ui container">
  <div
    className="ui grid">
    <div
      className="row">
      <div
        className="sixteen wide column">
        <div
          className="ui basic segment">
          <div
            className="ui breadcrumb">
            <a
              className="section"
              href="/accounts"
              onClick={[Function]}>
              Accounts
            </a>
            <div
              className="divider">
               / 
            </div>
            <div
              className="active section">
              Create a new account
            </div>
          </div>
        </div>
        <div
          className="ui basic segment">
          <form
            className="ui form"
            noValidate={true}
            onSubmit={[Function]}>
            <div
              className="ui header">
              Create an Account
            </div>
            <div
              className="field required">
              <label>
                Username
              </label>
              <div
                className="field">
                <div
                  className="ui input">
                  <input
                    name="username"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="" />
                </div>
              </div>
            </div>
            <div
              className="equal width fields">
              <div
                className="field">
                <label>
                  First Name
                </label>
                <div
                  className="field">
                  <div
                    className="ui input">
                    <input
                      name="first_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder=""
                      type="text"
                      value="" />
                  </div>
                </div>
              </div>
              <div
                className="field">
                <label>
                  Last Name
                </label>
                <div
                  className="field">
                  <div
                    className="ui input">
                    <input
                      name="last_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder=""
                      type="text"
                      value="" />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="field">
              <label>
                Roles
              </label>
              <div
                className="field">
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={true}
                  className="ui fluid multiple selection dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}>
                  <select
                    aria-hidden="true"
                    multiple={true}
                    name="roles"
                    type="hidden"
                    value={Array []}>
                    <option
                      value="" />
                    <option
                      value="admin">
                      Administrator
                    </option>
                    <option
                      value="containers:ro">
                      Containers Read-Only
                    </option>
                    <option
                      value="containers:rw">
                      Containers Read/Write
                    </option>
                    <option
                      value="images:ro">
                      Images Read-Only
                    </option>
                    <option
                      value="images:rw">
                      Images Read/Write
                    </option>
                    <option
                      value="nodes:ro">
                      Nodes Read-Only
                    </option>
                    <option
                      value="nodes:rw">
                      Nodes Read/Write
                    </option>
                    <option
                      value="registries:ro">
                      Registries Read-Only
                    </option>
                    <option
                      value="registries:rw">
                      Registries Read/Write
                    </option>
                    <option
                      value="events:ro">
                      Events Read-Only
                    </option>
                    <option
                      value="events:rw">
                      Events Read/Write
                    </option>
                  </select>
                  <div
                    className="default text">
                    Select user roles
                  </div>
                  <i
                    aria-hidden="true"
                    className="dropdown icon" />
                  <div
                    className="menu transition">
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={true}
                      className="selected item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Administrator
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Containers Read-Only
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Containers Read/Write
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Images Read-Only
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Images Read/Write
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Nodes Read-Only
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Nodes Read/Write
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Registries Read-Only
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Registries Read/Write
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Events Read-Only
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }>
                      <span
                        className="text">
                        Events Read/Write
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="field required">
              <label>
                Password
              </label>
              <div
                className="field">
                <div
                  className="ui input">
                  <input
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="password"
                    value="" />
                </div>
              </div>
            </div>
            <div
              className="field required">
              <label>
                Repeat Password
              </label>
              <div
                className="field">
                <div
                  className="ui input">
                  <input
                    name="passwordConfirm"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="password"
                    value="" />
                </div>
              </div>
            </div>
            <div
              className="ui hidden divider" />
            <div
              className="field">
              <button
                className="ui green button"
                onClick={[Function]}
                required={undefined}
                tabIndex={undefined}
                type={undefined}>
                Create Account
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
